presto-main/src/main/java/com/facebook/presto/type/ArrayType.java

```diff
@@ -86,8 +84,7 @@ public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int
     @Override
     public int hash(Block block, int position)
     {
-        Slice value = getSlice(block, position);
-        Block array = readStructuralBlock(value);
+        Block array = (Block) getObject(block, position);
```

Comment:
you can change `ArrayType.getObject` to return `Block` then you won't need this cast, or casts in other places that know they have an `ArrayType`